src/compile.coffee

```diff
@@ -1,45 +1,50 @@
-
 clone = require('clone')
 caseless = require('caseless')
 
 {child, children, parent, content} = require('./refract')
 validateParameters = require('./validate-parameters')
 detectTransactionExamples = require('./detect-transaction-examples')
 expandUriTemplateWithParameters = require('./expand-uri-template-with-parameters')
-apiElementsToJson = require('./api-elements-to-json')
+apiElementsToRefract = require('./api-elements-to-refract')
 
 
 compile = (mediaType, apiElements, filename) ->
-  parseResult = apiElementsToJson(apiElements)
-
   transactions = []
   errors = []
   warnings = []
 
-  component = 'apiDescriptionParser'
-  for annotation in children(parseResult, {element: 'annotation'})
-    group = if annotation.meta.classes[0] is 'warning' then warnings else errors
+  apiElements.annotations.forEach((annotation) ->
```

Comment:
Ha! Finally! I became so fluent in minim now that I can share some wisdom as well! 😜 Did you know you can do this with the `minim-api-description` namespace?

```coffeescript
errors = apiElements.errors.map(fn)
warnings = apiElements.warnings.map(fn)
```

And thanks for the suggestion, I like it. Feels cleaner code.